src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -0,0 +1,544 @@
+        /// Returns (n+(div-1))/div, rearranged to avoid arithmetic overflow. 
+        /// For example, in the bit to int case, the straightforward calc would 
+        /// be (n+31)/32, but that would cause overflow. So instead it's 
+        /// rearranged to ((n-1)/32) + 1, with special casing for 0.
+        /// 
+        /// Usage:
+        /// GetArrayLength(77, BitsPerInt32): returns how many ints must be 
+        /// allocated to store 77 bits.
+        /// </summary>
+        /// <param name="n"></param>
+        /// <param name="div">use a conversion constant, e.g. BytesPerInt32 to get
+        /// how many ints are required to store n bytes</param>
+        /// <returns></returns>
+        private static int GetArrayLength(int n, int div)
+        {
+            Contract.Assert(div > 0, "GetArrayLength: div arg must be greater than 0");
+            return n > 0 ? (((n - 1) / div) + 1) : 0;
+        }
+
+        private class BitArrayEnumeratorSimple : IEnumerator
+        {
+            private BitArray bitarray;
+            private int index;
+            private int version;
+            private bool currentElement;
+
+            internal BitArrayEnumeratorSimple(BitArray bitarray)
+            {
+                this.bitarray = bitarray;
+                this.index = -1;
+                version = bitarray._version;
+            }
+
+            public Object Clone()
+            {
+                return MemberwiseClone();
+            }
+
+            public virtual bool MoveNext()
+            {
+                ICollection bitarrayAsICollection = bitarray;
+                if (version != bitarray._version) throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
+                if (index < (bitarrayAsICollection.Count - 1))
+                {
+                    index++;
+                    currentElement = bitarray.Get(index);
+                    return true;
+                }
+                else
+                    index = bitarrayAsICollection.Count;
+
+                return false;
+            }
+
+            public virtual Object Current
+            {
+                get
+                {
+                    if (index == -1)
+                        throw new InvalidOperationException(SR.InvalidOperation_EnumNotStarted);
+                    if (index >= ((ICollection)bitarray).Count)
+                        throw new InvalidOperationException(SR.InvalidOperation_EnumEnded);
+                    return currentElement;
+                }
+            }
+
+            public void Reset()
+            {
+                if (version != bitarray._version) throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
+                index = -1;
+            }
+        }
+
+        private int[] m_array;
+        private int m_length;
+        private int _version;
+        private Object _syncRoot;
+
+        private const int _ShrinkThreshold = 256;
```

Comment:
Yep, thanks Stephen.  Updated, and filed https://github.com/dotnet/codeformatter/issues/117 .